File: src/AddAPearl.Domain/Company.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using AddAPearl.Core;

namespace AddAPearl.Domain
{
    public class Company : ICompany
    {
        public int CompanyId { get; set; }
        
        [Required]
        public string CompanyName { get; set; }

        public int? AddressId { get; set; }

        public IAddress Address { get; set; }
    }
}

File: AddAPearl.Web/src/app/shared/models/company.ts
import {
	IAddress,
} from './index';

/* Defines the company entity */
export interface ICompany {
	companyId: number;
	companyName: string;
	email: string;
	addressId: number;
	address: IAddress;
}

export class Company implements ICompany {
	constructor(
		public companyId: number,
		public companyName: string,
		public email: string,
		public addressId: number,
		public address: IAddress,
	) {
		// Any other stuff to do during construction...
	};
}
